services/datamanager/builtin/builtin.go

```diff
@@ -485,19 +464,23 @@ func (svc *builtIn) startSyncScheduler(intervalMins float64) {
 func (svc *builtIn) cancelSyncScheduler() {
 	if svc.syncRoutineCancelFn != nil {
 		svc.syncRoutineCancelFn()
+		svc.backgroundWorkers.Wait()
 		svc.syncRoutineCancelFn = nil
 	}
 }
 
 func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64) {
+	started := make(chan struct{})
```

Comment:
This was added to ensure this routine has started before this (and Update) return. It was necessary to avoid race conditions adding to `clock` (clock.Ticker and clock.Add cannot be called concurrently).